common/achievement/types.ts

```diff
@@ -41,10 +27,10 @@ export interface BucketEventsWithAggr extends BucketEvents {
     aggregation: EventValue;
 }
 
-type BucketFormulaFunction = (context: FormulaContext) => Bucket[];
+type BucketFormulaFunction<ID extends ActionID> = (context: SpecificNotificationContext<ID>) => Bucket[];
```

Comment:
@Jonasdoubleyou do we also need the builder for buckets to work with the `SpecificNotificationContext`?

Reply:
Does the bucket formula function need the full context? Or would the "relationID" be enough?